daemon/cluster/convert/service.go

```diff
@@ -198,6 +203,21 @@ func ServiceSpecToGRPC(s types.ServiceSpec) (swarmapi.ServiceSpec, error) {
 					},
 				},
 			}
+		} else {
+			return swarmapi.ServiceSpec{}, ErrMismatchedRuntime
+		}
+	case types.RuntimeNetworkAttachment:
+		// agree to parse this runtime type, but it's not actually valid to
+		// create a service with this runtime type. but that logic is someone
```

Comment:
whose responsibility is it? do we check for this situation? (because this looks to be the code you referred to to produce a proper error in https://github.com/moby/moby/pull/35246#issuecomment-340541386)

Reply:
ping @dperny ^^